### ISSUE TITLE
QueryNetwork breaks the network

### ISSUE THREAD

**Issue by leoll2** [openvinotoolkit/openvino#1819]
##### System information (version)

- OpenVINO=> 2020.2
- Operating System / Platform => Ubuntu 18.04
- Compiler => g++
- Problem classification => QueryNetwork bug

##### Detailed description

The method `InferenceEngine::Core::QueryNetwork` modifies the internal representation of the network silently, eventually causing troubles and crashes in other methods invoked later. I suspect the problem is due to a conversion from ngraph to the old representation, as described [here](https://docs.openvinotoolkit.org/2020.1/_docs_IE_DG_nGraph_Flow.html).
However, I didn't get any deprecation warning when compiling or at runtime, which makes this error very tricky and annoying.
Can you confirm this behavior is a bug, or at least should be flagged? Also, is there any alternative to `QueryNetwork` to get a list of the supported layers for a specific device?

##### Steps to reproduce

This code
```
auto net = ie_.ReadNetwork(cfgpath)
ie_.QueryNetwork(net, "MYRIAD")
auto ie_.LoadNetwork(net, "MYRIAD")     // -> FAIL
```
dies with the error: 
`  what():  Unexpected CNNNetwork format: it was converted to deprecated format prior plugin's call`

Instead, the same code without `QueryNetwork` runs fine.
```
auto net = ie_.ReadNetwork(cfgpath)
auto ie_.LoadNetwork(net, "MYRIAD")     // -> OK
```

Network model available at [this link](https://easyupload.io/9hzljl).

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [x] There is reproducer code and related data files: images, videos, models, etc.

**Comment by jgespino** [comment]
Hi @leoll2 

Did you try on the latest OpenVINO 2020.4 release? I was not able to reproduce your issue with your model and the following code:

```
from openvino.inference_engine import IENetwork, IECore

ie = IECore()
net = ie.read_network("barcode-1.xml", "barcode-1.bin")

ie.query_network(net, "MYRIAD")
ie.load_network(net, "MYRIAD")
```

Regards,
Jesus

**Comment by leoll2** [comment]
It works in 2020.4, is it a bug of 2020.2 then?
Do you know any alternative to QueryNetwork to retrieve a list of the supported layers for a specific device?

**Comment by jgespino** [comment]
Hi @leoll2 

It was likely a bug addressed in either 2020.3 or 2020.4 release. Aside from QueryNetwork, you can find which layers are supported by the plugins in the following documentation: https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Supported_Frameworks_Layers.html

Regards,
Jesus
